### ISSUE TITLE
Lineage Display problem

### ISSUE THREAD

**Issue by caiyt2001** [datahub-project/datahub#3965]
**Describe the bug**
execute：
`python metadata-ingestion/examples/library/lineage_dataset_job_dataset.py`

db data：
`{
    "inputDatajobs":[
        "urn:li:dataJob:(urn:li:dataFlow:(airflow,flow1,PROD),job0)"
    ],
    "inputDatasets":[
        "urn:li:dataset:(urn:li:dataPlatform:mysql,librarydb.member,PROD)",
        "urn:li:dataset:(urn:li:dataPlatform:mysql,librarydb.checkout,PROD)"
    ],
    "outputDatasets":[
        "urn:li:dataset:(urn:li:dataPlatform:kafka,debezium.topics.librarydb.member_checkout,PROD)"
    ]
}`
but Front end display：
![image](https://user-images.githubusercontent.com/35490928/150903387-c86fb7d9-bbbc-41ce-aeac-c009c6a35529.png)

version：0.8.23
Run As：docker  quickstart.sh

**Comment by darapuk** [comment]
We're seeing this bug in our deployment as well, lineage seems to flicker between what's expected and other random tables if refreshed enough times

**Comment by darapuk** [comment]
Sample code to reproduce this bug:

```
import datahub.emitter.mce_builder as builder
from datahub.emitter.rest_emitter import DatahubRestEmitter

PLATFORM = "bigquery"

# Create an emitter to the GMS REST API.
emitter = DatahubRestEmitter("http://localhost:8080")

lineage_mce_l1 = builder.make_lineage_mce(
    [
        builder.make_dataset_urn(PLATFORM, "upstream_l1_table_1"),
        builder.make_dataset_urn(PLATFORM, "upstream_l1_table_2"),
    ],
    builder.make_dataset_urn(PLATFORM, "downstream"),
)

emitter.emit_mce(lineage_mce_l1)

lineage_mce_l2 = builder.make_lineage_mce(
    [
        builder.make_dataset_urn(PLATFORM, "upstream_l2_table_1"),
        builder.make_dataset_urn(PLATFORM, "upstream_l2_table_2"),
    ],
    builder.make_dataset_urn(PLATFORM, "upstream_l1_table_1"),
)

emitter.emit_mce(lineage_mce_l2)

lineage_mce_l0 = builder.make_lineage_mce(
    [
        builder.make_dataset_urn(PLATFORM, "downstream")
    ],
    builder.make_dataset_urn(PLATFORM, "downstream_l0"),
)

emitter.emit_mce(lineage_mce_l0)

## Expected:
# "upstream_l2_table_1" >> "upstream_l1_table_1"
# "upstream_l2_table_2" >> "upstream_l1_table_1"
# "upstream_l1_table_1" >> "downstream"
# "upstream_l1_table_2" >> "downstream"
```

**Comment by darapuk** [comment]
After refreshing a couple of times you see lineage break:
<img width="840" alt="Screen Shot 2022-01-28 at 12 27 36 PM" src="https://user-images.githubusercontent.com/49173956/151616919-22bacca2-88d3-4d06-b3c0-a33b1d684103.png">


Expected lineage:
<img width="1024" alt="Screen Shot 2022-01-28 at 12 28 04 PM" src="https://user-images.githubusercontent.com/49173956/151616954-c5ff8c33-6668-4cde-8beb-f0f617a78b88.png">


sorry in advance for the poor naming!

**Comment by jjoyce0510** [comment]
Thanks for the report! We are looking into this

**Comment by jjoyce0510** [comment]
This should have been resolved in a few days back before the release 0.8.24. This PR: https://github.com/linkedin/datahub/pull/3950 Please try to update and let us know if the issue persists. Thanks!

**Comment by darapuk** [comment]
Confirming that this works on master. Thanks for the tip @jjoyce0510! 

I think this can be closed.
